CMakeLists.txt

```diff
@@ -1683,7 +1683,22 @@ add_definitions("-DWOLFSSL_IGNORE_FILE_WARN")
 
 # Generate user options header
 message("Generating user options header...")
-set(OPTION_FILE "${CMAKE_CURRENT_SOURCE_DIR}/wolfssl/options.h")
+if (${CMAKE_DISABLE_SOURCE_CHANGES})
+    set(WOLFSSL_BUILD_OUT_OF_TREE_DEFAULT "${CMAKE_DISABLE_SOURCE_CHANGES}")
+else()
+    set(WOLFSSL_BUILD_OUT_OF_TREE_DEFAULT "no")
+endif()
+add_option("WOLFSSL_BUILD_OUT_OF_TREE"
+    "Don't generate files in the source tree (default: ${CMAKE_DISABLE_SOURCE_CHANGES})"
```

Comment:
I think this is functionally the same, but I would change `CMAKE_DISABLE_SOURCE_CHANGES` here to `WOLFSSL_BUILD_OUT_OF_TREE_DEFAULT`.

Reply:
Done.